### ISSUE TITLE
Warrior: Slam and Execute not in Combat Log when not landing

### ISSUE THREAD

**Issue by taralej90** [dalaranwow/dalaran-wow#2233]
When Slam and Execute don't land (Miss/Dodge/Parry), they do not show on the screen and they also do not enter the Combat Log.

Note: when Slam and Execute get Blocked, they do enter the Combat Log as Blocking counts as landed but mitigated.

**Comment by n-ghost** [comment]
Confirmed, this might mislead players into believing that sometimes Execute does no damage.

**Comment by taralej90** [comment]
Eh, is there any development on this issue? People keep asking why their Execute keeps disappearing.

**Comment by Vaxsysl** [comment]
You fail to mention that rage is still used up, while nothing happens.

**Comment by Aridam** [comment]
Confirming this as described originally, at least with Execute (don't have Slam yet) -- at times upon using Execute the sound is heard, Rage is used, animation triggers, but no Damage is actually dealt, resulting in a full waste.

**Comment by Aridam** [comment]
Reposting to give an update on this and request a revival of this alongside some testing being necessary.

The Execute/Slam "Misses" that are unreported actually _do_ sometimes end up as Misses/Avoids (noticed _some_ unreported Execute "Misses" gaining me an Overpower proc ergo it must've been an unreported Dodge), but the core nature of this is still tentative in my eyes (Rage depletion feels odd whenever this happens which is difficult to surely-tell, as well as these "Misses/Avoids" seeming too frequent to be true); will need someone to test with 100% Hit Chance and no Avoids possible while tracking Health/Rage/Combat Log/etc. to deduct whether or not these missing reports are merely just legitimate Misses/Avoids that somehow end up being omitted in reports when in fact they should _all be reported_, or if it's that _but also_ Misses/Avoids that _aren't even supposed to happen_ at times.

Currently I personally cannot test this with certainty as my circumstances don't allow me that in-game, which is why I'm calling upon someone who can to do so -- extensively, and with complete certainty.

**Comment by n-ghost** [comment]
Afair when I was testing this 2 years ago (>,>), getting Hit/Expertise capped would fix the issue. Either that or I really wished it was.

(i.e., I remember this being a thing constantly while leveling; I don't remember it at all while in raid and all oiled up)

**Comment by n-ghost** [comment]
Will just throw some wild useless thoughts here: 

- can one do glancing Slams/Executes right now? Do they enter combat log properly?

- does the issue occur with Bloodsurge Slams as well? Maybe with them only but not with the regular ones, or vice versa?

- could these missing attacks be tracked by any addon?

- are all ranks of abilities in question affected?

- are they affected by any rage cost modifiers, both positive or negative? I vaguely remember something fishy about this particular point.

 (i.e. it may be so that, for example, with rage cost reduced by 1 with Focused Rage, Execute becomes available at 14 rage, but server still thinks you need 15, and so if your attack fails to land it never enters the combat log because no damage was generated and you supposedly couldn't use the ability anyway.) 

- does it occur in conjuction with (talented) Heroic Strike/Cleave in queue? (same logic as above)

**Comment by Aridam** [comment]
The main reason as to why I'm still questioning the legitimacy of this bug is exactly the possibility of any hits or at least those coming off as Glancing to be omitted, and how true those may be (because sometimes the HP of my Target at the time failed to budge, yet Rage seemed to be fully-expended by Execute/Slam during the fail -- even Sudden Death would be consumed in the process -- hinting towards either actually having landed rather than Missed/Avoided), so that should receive some additional testing -- preferably by someone on the PTR for accessibility. My idea in that regard is to test once with full Crit, Hit, and Expertise -- in other words, to land _only_ Crit Slam/Execute hits. Then, test the same, _but_ the Crits -- to see how Glancing Blows register instead.

It _is_ confirmed that some Avoids really _are being omitted_ -- if at times you pay attention to this occurrence, you may hear a Parry sound which corresponds and seems legitimate. At other times I've even got valid Dodges that simply didn't get reported -- because Overpower procced from that, and that's how I noticed and established that fact. I have not yet witnessed omitted Blocks though (haven't heard any Block sounds or anything else in its regard to correspond to this anomaly occurring -- might just be my luck with it so far though).

The Misses are another tentative thing in my eyes -- sometimes it just feels like Slam/Execute get those Misses a bit too many times for comfort (ergo something not derived from chance just going that way), most of which are from the unreported ones (ergo no evident Parries/Dodges); can't be too certain. This should be the second thing to test extensively and deduct after the Crit/Hit/Glancing subject.

Personally I still am to this day unable to provide solid practical testing data as I'm not even L80 yet, much less able to get Hit- and Expertise-capped, but this happens to be a very-old bug that I know of with which I have some experience, so those are my two cents in its regard -- hope it helps.

Additionally, to answer the above-asked questions:

1. Unable to give the verdict on Bloodsurge Slams as I'm Arms and haven't tested with the Talent; regular Slam however _does_ act that way;

2. I believe these omitted reports are untrackable -- using Combat Log _and_ MSBT provided me with nothing in that department;

3. So far I've noticed it happening across the board for me (both Abilities), so it must be true that practically all Ranks are affected by this;

4. I'm not using any Rage-cost-modifying Talents so I won't give the verdict on this specific one either, but due to your suspicion being there, it might as well be added to the list of tests needed to be run just in case;

5. Same as just-above applies to HS/Cleave with Talents from me -- might want to consider that as well I suppose; however, I've deducted that at least with the regular HS/Cleaves queued and going off in-sync with Executes/Slams, it doesn't really differ, so I'd say no.

One could assume that with so many modifiers and possibilities just in general that may or may not alter how this occurs or doesn't, multiple tests in various subjects should be made to see how Execute/Slam behave with these differences taken into consideration -- to deduct whether or not any of this plays an additional role in the bug itself; if not, then it's just how they register Misses/Avoids/Glances/Hits/Crits, and how actually-legit some of those omitted Misses/Avoids that don't seem to budge HP yet expend Rage/SD procs are.

In other words, if this practically ends up as a false alarm and everything is legit, then there's nothing hurtful -- just some seemingly-random omits in Combat reports for these two Abilities; ergo a low-priority fix. If it's otherwise, though, then it could have serious implications.

**Comment by Aridam** [comment]
Coming back to this to reconfirm it still happens.

**Comment by Aridam** [comment]
Still an issue (also still believe this relates to the referenced issue).

**Comment by Aridam** [comment]
Still an issue and still related to the referenced issue with Recklessness -- please check.

**Comment by Aridam** [comment]
Bump.

Still present. Seems to have a second side to it that's harmful in every way and not just a mere visual.
